common/chat.ts

```diff
@@ -199,3 +241,32 @@ export interface ModerateChannelUserServerRequest {
    */
   moderationReason?: string
 }
+
+/**
+ * Specific chat information about a user in a particular channel, such as their join date, their
+ * channel role (e.g. are they a moderator), etc.
+ */
+export interface ChatUserProfile {
```

Comment:
imo these structures should usually always include the relevant ID as well (in this case, probably both the user ID and the channel name)